### PR TITLE
Migrate build.gradle to kotlin

### PR DESCRIPTION
**Pull request by anuraaga**
Did this first since it's the doozy. The code's too different to have any reasonable diff with the groovy so just going to have to eyeball I think.

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,559 @@
...
+import net.ltgt.gradle.errorprone.ErrorPronePlugin
+import org.gradle.api.plugins.JavaPlugin.*
+import org.gradle.api.tasks.testing.logging.TestExceptionFormat
+import org.jfrog.gradle.plugin.artifactory.dsl.ArtifactoryPluginConvention
+import org.jfrog.gradle.plugin.artifactory.dsl.PublisherConfig
+import org.jfrog.gradle.plugin.artifactory.dsl.ResolverConfig
+import ru.vyarus.gradle.plugin.animalsniffer.AnimalSniffer
+import ru.vyarus.gradle.plugin.animalsniffer.AnimalSnifferPlugin
+
+plugins {
```
Comment (jkwatson) [inline build.gradle.kts:19]: would it make sense to have 2 `plugins` closures? One for the things we apply universally and one for the bits we don't?

Comment (jkwatson) [reply build.gradle.kts:19]: hm. looks like we can't have 2 of them, but maybe we can separate the `apply false` from the others, just for clarity?

File: build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,559 @@
...
+}
+
+if (!JavaVersion.current().isJava11Compatible()) {
+    throw GradleException("JDK 11 or higher is required to build. " +
+            "One option is to download it from https://adoptopenjdk.net/. If you believe you already " +
+            "have it, please check that the JAVA_HOME environment variable is pointing at the " +
+            "JDK 11 installation.")
+}
+
+// Nebula plugin will not configure if .git doesn't exist, let's allow building on it by stubbing it
```
Comment (jkwatson) [inline build.gradle.kts:39]: how plausible will it be, now that we're using kotlin, to do some refactoring of the build script to make it more modular? Can we move pieces into files that are appropriately named to make it really easy to find the pieces related to publishing, for example?

File: build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,559 @@
...
+                    allDisabledChecksAsWarnings.set(true)
+
+                    // Doesn't currently use Var annotations.
+                    disable("Var") // "-Xep:Var:OFF"
+
+                    // ImmutableRefactoring suggests using com.google.errorprone.annotations.Immutable,
+                    // but currently uses javax.annotation.concurrent.Immutable
+                    disable("ImmutableRefactoring") // "-Xep:ImmutableRefactoring:OFF"
+
+
```
Comment (jkwatson) [inline build.gradle.kts:164]: there are a couple of extra CR/LFs scattered in here. on purpose, or can we clean this up?

File: build.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,559 @@
...
+                                license {
+                                    name.set("The Apache License, Version 2.0")
+                                    url.set("http://www.apache.org/licenses/LICENSE-2.0.txt")
+                                }
+                            }
+
+                            developers {
+                                developer {
+                                    id.set("opentelemetry")
+                                    name.set("OpenTelemetry Gitter")
```
Comment (jkwatson) [inline build.gradle.kts:458]: we should definitely change this! [doesn't have to be in this PR]

**Review by jkwatson — APPROVED** [review]
less groovy makes me more happy. 
👍

File: build.gradle.kts (hunk 1 above)
Comment (anuraaga) [reply build.gradle.kts:19]: Good idea

File: build.gradle.kts (hunk 2 above)
Comment (anuraaga) [reply build.gradle.kts:40]: Yup I think we can do some splitting soon

**Review by anuraaga — COMMENTED** [review]
Tried publishToMavenLocal with this PR and see sources jar. Weird, will see if that holds for snapshots too wouldn't expect a change though.

**Pull request merged**